test/uk/gov/hmrc/eusubsidycompliancefrontend/services/EmailVerificationServiceSpec.scala

```diff
@@ -0,0 +1,163 @@
+import uk.gov.hmrc.mongo.cache.{CacheItem, DataKey}
+import uk.gov.hmrc.mongo.test.DefaultPlayMongoRepositorySupport
+
+import java.time.Instant
+import scala.concurrent.Await
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.Duration
+
+class EmailVerificationServiceSpec extends AnyWordSpec with Matchers with BeforeAndAfterAll with MockFactory with ScalaFutures with DefaultAwaitTimeout with DefaultPlayMongoRepositorySupport[CacheItem] {
+
+
+  implicit val hc: HeaderCarrier = mock[HeaderCarrier]
+  private val mockEmailVerificationConnector: EmailVerificationConnector = mock[EmailVerificationConnector]
+  implicit val writes: Writes[EmailVerificationState] = Json.writes[EmailVerificationState]
+
+    override protected def repository = new EoriEmailDatastore(mongoComponent)
+
+    private val service = new EmailVerificationService(
+    mockEmailVerificationConnector,
+    repository
+  )
+  override def afterAll(): Unit = {
+    repository.collection.deleteMany(filter = Filters.exists("_id"))
+  }
+
+  val unverifiedVerificationRequest = EmailVerificationState("unverified@something.com", Some("someId"), false.some)
+  val verifiedVerificationRequest = EmailVerificationState("verified@something.com", Some("someId"), true.some)
+
+  val mockVerifyEmailResponse = VerifyEmailResponse("testRedirectUrl")
+
+
+  "EmailVerificationService" when {
+
+    "getEmailVerification is called" must {
+
+      "No verification found" when {
+
+        "return none" in {
+          Await.result(repository.put(eori1, unverifiedVerificationRequest), Duration(10, "seconds"))
+          service.getEmailVerification(eori1).futureValue shouldBe None
+        }
+      }
+
+      "Verification found" when {
+
+        "return verified" in {
+          Await.result(repository.put(eori2, verifiedVerificationRequest), Duration(10, "seconds"))
+          service.getEmailVerification(eori2).futureValue shouldBe verifiedVerificationRequest.some
+        }
+      }
+
+    }
+
+    "verifyEori is called" must {
+
+      "OK" when {
+
+        "record is verified" in {
+          Await.result(repository.put(eori3, unverifiedVerificationRequest), Duration(10, "seconds"))
+          service.getEmailVerification(eori3).futureValue shouldBe None
+          service.verifyEori(eori3)
+          service.getEmailVerification(eori3).futureValue shouldBe unverifiedVerificationRequest.copy(verified = true.some).some
+        }
+
+      }
+    }
+
+    "verifyVerificationRequest is called" must {
+
+      "verify record" when {
+        "success" in {
+          Await.result(repository.put(eori1, unverifiedVerificationRequest.copy(pendingVerificationId = "pending".some)), Duration(10, "seconds"))
+          service.verifyVerificationRequest(eori1, "pending")
+          service.getEmailVerification(eori1).futureValue shouldBe unverifiedVerificationRequest.copy(verified = true.some, pendingVerificationId = "pending".some).some
```

Comment:
If I run the entire suite locally I get a failure on this test as follows

```
[info]   when verifyVerificationRequest is called
[info]     must verify record
[info]     - when success *** FAILED ***
[info]       None was not equal to Some(EmailVerificationState(unverified@something.com,Some(pending),Some(true))) (EmailVerificationServiceSpec.scala:120)
```

It seems to work if I run the test in isolation. Might be worth looking into if you get this behaviour. Likely I'm missing something somewhere!